File: rgctl-tests/ecommerce-go/internal/models/review.go
package models

type Review struct {
	ID        string `gorm:"primaryKey" json:"id"`
	ProductID string `gorm:"not null;index" json:"product_id"`
	UserID    string `gorm:"not null" json:"user_id"`
	Rating    int64  `gorm:"not null" json:"rating"`
	Comment   string `gorm:"not null;default:''" json:"comment"`
	CreatedAt string `gorm:"not null" json:"created_at"`
}

func (r *Review) SetCreatedAt(v string) { r.CreatedAt = v }

File: rgctl-tests/ecommerce-rust/src/repositories/review.rs
use crate::{error::AppResult, models::review::Review};
use sqlx::SqlitePool;

pub async fn create(pool: &SqlitePool, r: &Review) -> AppResult<Review> {
    sqlx::query_as::<_, Review>(
        "INSERT INTO reviews (id, product_id, user_id, rating, comment, created_at) VALUES (?, ?, ?, ?, ?, ?) RETURNING id, product_id, user_id, rating, comment, created_at",
    )
    .bind(&r.id)
    .bind(&r.product_id)
    .bind(&r.user_id)
    .bind(r.rating)
    .bind(&r.comment)
    .bind(&r.created_at)
    .fetch_one(pool)
    .await
    .map_err(Into::into)
}

pub async fn list_for_product(pool: &SqlitePool, product_id: &str) -> AppResult<Vec<Review>> {
    sqlx::query_as::<_, Review>(
        "SELECT id, product_id, user_id, rating, comment, created_at FROM reviews WHERE product_id = ? ORDER BY created_at DESC",
    )
    .bind(product_id)
    .fetch_all(pool)
    .await
    .map_err(Into::into)
}
